std/assembly/rt/tlsf.ts

```diff
@@ -379,6 +379,10 @@ function prepareBlock(root: Root, block: Block, size: usize): void {
 /** Adds more memory to the pool. */
 function addMemory(root: Root, start: usize, endU64: u64): bool {
   let end = <usize>endU64;
+  if (ASC_LOW_MEMORY_LIMIT) {
+    end = <u64>ASC_LOW_MEMORY_LIMIT;
+    if (start > ASC_LOW_MEMORY_LIMIT) unreachable();
+  }
```

Comment:
@HerrCai0907 would this work? Also, what's the background behind overwriting `endU64`?
```suggestion
  if (ASC_LOW_MEMORY_LIMIT) endU64 = ASC_LOW_MEMORY_LIMIT;
  let end = <usize>endU64;
```

Reply:
The original implement is u32, but huge memory will cause u32 overflow and failed the later assert.
I don't want to mix the debug assert (it should not happen) and oom (it may happen).

Reply:
I don't see how that overflow could happen. Memories larger than 4 GiB only exist when using memory64, and in that case, `usize` would be equivalent to `u64`, not `u32` (so no overflow could take place).

Also, about separating those assertions, you can just change the `if (DEBUG)` on the next line to `if (ASC_LOW_MEMORY_LIMIT || DEBUG)`.

One last thing: **can't `ASC_LOW_MEMORY_LIMIT` be larger than the current memory size in bytes?** An instance could start with a memory of 2 pages, but `ASC_LOW_MEMORY_LIMIT` could be set to 224 KiB (3 and a half pages). In `initialize` and `computeSize`, `addMemory` is called with an `endU64` argument of `memory.size() << 16`, which is the number of bytes available to the instance (if we assume every page has been committed). However, with this change, the value of `end` is changed in `addMemory` from the given value (`memory.size() << 16`) to `ASC_LOW_MEMORY_LIMIT`. While this wouldn't have been a problem before, since previously this was only supposed to work with amounts smaller than 1 page, `ASC_LOW_MEMORY_LIMIT` could be larger than `memory.size() << 16`, and that could lead to `addMemory` adding memory that the instance doesn't have yet (and in turn, cause a crash due to an OOB memory access when someone tries to allocate there).

Reply:
the key point is 4gb, it will overflow to zero